pkg/operator/config.go

```diff
@@ -166,6 +174,8 @@ func getProviderControllerFromImages(provider Provider, images Images) (string,
 		return images.ClusterAPIControllerOpenStack, nil
 	case KubemarkProvider:
 		return images.ClusterAPIControllerKubemark, nil
+	case VSphereProvider:
+		return "vSphere", nil
```

Comment:
Any reason not to follow the existing convention of `images.ClusterAPIController<Type>`?

Reply:
This should also use the existing constant:

```diff
--- a/pkg/operator/config.go
+++ b/pkg/operator/config.go
@@ -175,7 +175,7 @@ func getProviderControllerFromImages(provider Provider, images Images) (string,
        case KubemarkProvider:
                return images.ClusterAPIControllerKubemark, nil
        case VSphereProvider:
-               return "vSphere", nil
+               return string(VSphereProvider), nil
        case BareMetalProvider:
                return images.ClusterAPIControllerBareMetal, nil
        case NoneProvider:
```

Reply:
Oh - but this is "vSphere" and the Provider constant is "vsphere". Which wins?

Reply:
the images values comes from a configMap which images are replaced by automation when building the release payload based on the mapping imageStream here https://github.com/openshift/machine-api-operator/blob/master/install/image-references
So hardcoding here for those not wanting machine API controllers/images is fine

Reply:
I was following the convention laid done by the None provider. That provider uses lowercase for the provider name and mixed case for the controller name. The ofiicial name of the vSphere product uses a lowercase for the v, so that is why I choose vSphere over VSphere.

Reply:
Resolved by cd6ac6f...3844ccf